experimenter/experimenter/features/manifests/nimbus_fml_loader.py

```diff
@@ -14,26 +14,18 @@ class NimbusFmlLoader:
     BASE_PATH = Path(BASE_DIR, "features", "manifests", "apps.yaml")
     MANIFEST_PATH = Path(BASE_DIR, "features", "manifests")
 
-    def __init__(self, application: str, channel: str, file_location=BASE_PATH):
+    def __init__(self, application: str, channel: str):
         self.application: str = (
             application if application in NimbusConstants.Application else None
         )
         self.channel: str = channel if channel in NimbusConstants.Channel else None
-        self.application_data = self._get_application_data(application, file_location)
 
     @staticmethod
-    def _get_application_data(application_name, file_location=BASE_PATH):
-        """Fetch app data from local apps.yaml file to find application names, github
-        repos, fml paths within the repos, and release versions.
-        """
-        if Path.exists(file_location):
-            with open(file_location) as application_yaml_file:
-                file = yaml.safe_load(application_yaml_file.read())
-                if application_name in file:
-                    return file[application_name]
-                else:
-                    return None
+    @cache
+    def create_loader(application: str, channel: str):
+        return NimbusFmlLoader(application, channel)
 
+    @cache  # noqa: B019
     def _get_local_file_path(self):
```

Comment:
We should used [`cached_property`](https://docs.python.org/3/library/functools.html#functools.lru_cache) here because there is only ever one value.

If we switch it to a property, it should be named as such (i.e., no `get_`).

```suggestion
    @cached_property
    def file_path(self):
```

Reply:
It might not be a single value, because the user might change the `Channel` (since the Audience page comes after the Branches page). Does `cached_property` still make sense if it can change like that?

Reply:
I decided to remove the caching on this altogether, since it really won't change 👍